mlrun/runtimes/function.py

```diff
@@ -1223,6 +1223,50 @@ def resolve_function_http_trigger(function_spec):
         return trigger_config
 
 
+def _resolve_function_image_pull_secret(function):
+    """
+    the corresponding attribute for 'build.secret' in nuclio is imagePullSecrets, attached link for reference
+    https://github.com/nuclio/nuclio/blob/e4af2a000dc52ee17337e75181ecb2652b9bf4e5/pkg/processor/build/builder.go#L1073
+    if only 1 of the secrets is set, use it.
+    if both are set and different, use the non default one and give precedence to image_pull_secret
+    """
+    # enrich with defaults
+    if function.spec.image_pull_secret is None:
+        function.spec.image_pull_secret = (
+            mlrun.mlconf.function.spec.image_pull_secret.default
+        )
+
+    if function.spec.build.secret is None:
+        function.spec.build.secret = mlrun.mlconf.httpdb.builder.docker_registry_secret
+
+    image_pull_secret = function.spec.image_pull_secret or function.spec.build.secret
+    if (
+        function.spec.image_pull_secret
+        and function.spec.build.secret
+        and function.spec.image_pull_secret != function.spec.build.secret
+    ):
+        message = "Image pull secret and build secret should match for nuclio runtime"
+        message_usage_info = ", using image pull secret."
+        if (
+            function.spec.build.secret
+            != mlrun.mlconf.httpdb.builder.docker_registry_secret
+            and function.spec.image_pull_secret
+            == mlrun.mlconf.function.spec.image_pull_secret.default
+        ):
+            # build secret was set by user, use it
+            image_pull_secret = function.spec.build.secret
+            message_usage_info = ", using build secret."
+
+        message += message_usage_info
+        logger.warning(
+            message,
+            image_pull_secret=function.spec.image_pull_secret,
+            build_secret=function.spec.build.secret,
+        )
+
```

Comment:
note that this warning won't get to the user, as this code is expected to run on server, so I suggest removing it